spec/functions/fqdn_rotate_spec.rb

```diff
@@ -68,9 +66,9 @@ def fqdn_rotate(value, args = {})
 
     # workaround not being able to use let(:facts) because some tests need
     # multiple different hostnames in one context
-    allow(scope).to receive(:lookupvar).with('::fqdn').and_return(host)
+    allow(subject.func.closure_scope).to receive(:[]).with('facts').and_return({ 'networking' => { 'fqdn' => host } })
```

Comment:
What I really wanted to do was mock the function's private method `fqdn_fact`, but can't figure out how to do this.